### PR TITLE
Enable noctx linter and fix up issues

### PR DESCRIPTION
**Pull request by alexandear**
This PR enables [`noctx`](https://github.com/sonatard/noctx) linter in golangci-lint's config. The linter requires the usage of HTTP requests with context for cancelation long-running requests.

### CODE REVIEW

File: pkg/downloader/downloader.go [hunk 1]
```diff
@@ -173,7 +175,7 @@ func Download(local, remote string, opts ...Opt) (*Result, error) {
 	}
 
 	if o.cacheDir == "" {
-		if err := downloadHTTP(localPath, remote, o.description, o.expectedDigest); err != nil {
+		if err := downloadHTTP(context.TODO(), localPath, remote, o.description, o.expectedDigest); err != nil {
```
Comment (AkihiroSuda) [inline pkg/downloader/downloader.go:178]: ctx should be passed from the caller of `Download()`

File: pkg/networks/usernet/client.go [hunk 2]
```diff
@@ -95,14 +96,12 @@ func (c *Client) ResolveIPAddress(vmMacAddr string) (string, error) {
 }
 
 func (c *Client) Leases() (map[string]string, error) {
-	res, err := c.client.Get(fmt.Sprintf("%s%s", c.base, "/services/dhcp/leases"))
+	u := fmt.Sprintf("%s%s", c.base, "/services/dhcp/leases")
+	res, err := httpclientutil.Get(context.TODO(), c.client, u)
```
Comment (AkihiroSuda) [inline pkg/networks/usernet/client.go:100]: ctx should be passed from the caller of `Leases()`

File: pkg/networks/usernet/client.go [hunk 3]
```diff
@@ -95,14 +96,12 @@ func (c *Client) ResolveIPAddress(vmMacAddr string) (string, error) {
...
 	if err != nil {
 		return nil, err
 	}
 	defer res.Body.Close()
-	if res.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("unexpected status: %d", res.StatusCode)
-	}
```
Comment (AkihiroSuda) [inline pkg/networks/usernet/client.go:105]: Why remove?

File: pkg/downloader/downloader.go [hunk 4]
```diff
@@ -487,14 +489,11 @@ func downloadHTTP(localPath, url, description string, expectedDigest digest.Dige
...
 
-	resp, err := http.Get(url)
+	resp, err := httpclientutil.Get(ctx, http.DefaultClient, url)
 	if err != nil {
 		return err
 	}
 	defer resp.Body.Close()
-	if resp.StatusCode != http.StatusOK {
-		return fmt.Errorf("expected HTTP status %d, got %s", http.StatusOK, resp.Status)
-	}
```
Comment (AkihiroSuda) [inline pkg/downloader/downloader.go:497]: Why remove?

File: pkg/networks/usernet/client.go (hunk 3 above)
Comment (alexandear) [reply pkg/networks/usernet/client.go:105]: `httpclientutil.Get` already checks for successful status code, see https://github.com/lima-vm/lima/blob/a00bd52f78f86fc611f7c2a098597ae2892762ba/pkg/httpclientutil/httpclientutil.go#L28

File: pkg/downloader/downloader.go (hunk 4 above)
Comment (alexandear) [reply pkg/downloader/downloader.go:497]: `httpclientutil.Get` already checks for successful status code, see https://github.com/lima-vm/lima/blob/a00bd52f78f86fc611f7c2a098597ae2892762ba/pkg/httpclientutil/httpclientutil.go#L28

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**